### PR TITLE
avahi: update to 0.8 and dependencies

### PR DESCRIPTION
**Pull request by heitbaum**

### CODE REVIEW

File: packages/network/avahi/package.mk [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 PKG_NAME="avahi"
-PKG_VERSION="0.7"
-PKG_SHA256="fd45480cef0559b3eab965ea3ad4fe2d7a8f27db32c851a032ee0b487c378329"
+PKG_VERSION="0.8"
+PKG_SHA256="c15e750ef7c6df595fb5f2ce10cac0fee2353649600e6919ad08ae8871e4945f"
 PKG_LICENSE="GPL"
 PKG_SITE="http://avahi.org/"
 PKG_URL="https://github.com/lathiat/avahi/archive/v$PKG_VERSION.tar.gz"
-PKG_DEPENDS_TARGET="toolchain expat libdaemon dbus connman"
+PKG_DEPENDS_TARGET="toolchain expat libdaemon dbus connman libevent gettext"
```
Comment (lrusak) [inline packages/network/avahi/package.mk:11]: do we need libevent or can you just disable it with `--disable-libevent`??

Comment (heitbaum) [reply packages/network/avahi/package.mk:11]: Will do a code review and some tests.

**Comment by CvH** [comment]
```
CMake Error at /LE/build.LibreELEC-Generic.x86_64-9.80-devel/toolchain/share/cmake-3.19/Modules/FindPackageHandleStandardArgs.cmake:218 (message):
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY
  OPENSSL_INCLUDE_DIR)
Call Stack (most recent call first):
  /LE/build.LibreELEC-Generic.x86_64-9.80-devel/toolchain/share/cmake-3.19/Modules/FindPackageHandleStandardArgs.cmake:582 (_FPHSA_FAILURE_MESSAGE)
  /LE/build.LibreELEC-Generic.x86_64-9.80-devel/toolchain/share/cmake-3.19/Modules/FindOpenSSL.cmake:536 (find_package_handle_standard_args)
  CMakeLists.txt:844 (find_package)


-- Configuring incomplete, errors occurred!
See also "/LE/build.LibreELEC-Generic.x86_64-9.80-devel/build/libevent-2.1.12-stable/.x86_64-libreelec-linux-gnu/CMakeFiles/CMakeOutput.log".
See also "/LE/build.LibreELEC-Generic.x86_64-9.80-devel/build/libevent-2.1.12-stable/.x86_64-libreelec-linux-gnu/CMakeFiles/CMakeError.log".
FAILURE: scripts/build libevent:target during configure_target (default)
*********** FAILED COMMAND ***********
cmake ${CMAKE_GENERATOR_NINJA} ${TARGET_CMAKE_OPTS} ${PKG_CMAKE_OPTS_TARGET} $(dirname ${PKG_CMAKE_SCRIPT})
**************************************
*********** FAILED COMMAND ***********
${SCRIPTS}/build "${1}" "${PARENT_PKG}"
**************************************
FAILURE: scripts/install libevent:target has failed!
```

adding openssl to deps fixed it

File: packages/network/libevent/package.mk [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# SPDX-License-Identifier: GPL-2.0
+# Copyright (C) 2020-present Team LibreELEC (https://libreelec.tv)
+
+PKG_NAME="libevent"
+PKG_VERSION="2.1.12-stable"
```
Comment (CvH) [inline packages/network/libevent/package.mk:5]:
```suggestion
PKG_VERSION="2.1.12"
```

File: packages/network/libevent/package.mk [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+PKG_SHA256="92e6de1be9ec176428fd2367677e61ceffc2ee1cb119035037a27d346b0403bb"
+PKG_LICENSE="BSB-3c"
+PKG_SITE="https://github.com/libevent/libevent"
+PKG_URL="https://github.com/libevent/libevent/releases/download/release-$PKG_VERSION/libevent-$PKG_VERSION.tar.gz"
```
Comment (CvH) [inline packages/network/libevent/package.mk:9]:
```suggestion
PKG_URL="https://github.com/libevent/libevent/releases/download/release-$PKG_VERSION/libevent-$PKG_VERSION-stable.tar.gz"
```

File: packages/network/libevent/package.mk [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+PKG_LONGDESC="Event notification library"
+PKG_DEPENDS_TARGET="toolchain"
+PKG_TOOLCHAIN="cmake"
+
+PKG_CMAKE_OPTS_TARGET="-DCMAKE_CXX_STANDARD=14 -DCMAKE_CXX_EXTENSIONS:BOOL=OFF -DSPDLOG_FMT_EXTERNAL=ON -DSPDLOG_BUILD_EXAMPLE=OFF -DSPDLOG_BUILD_TESTS=OFF"
```
Comment (CvH) [inline packages/network/libevent/package.mk:14]:
```suggestion
PKG_CMAKE_OPTS_TARGET="-DCMAKE_CXX_EXTENSIONS:BOOL=OFF \
                       -DCMAKE_CXX_STANDARD=14 \
                       -DSPDLOG_BUILD_EXAMPLE=OFF \
                       -DSPDLOG_BUILD_TESTS=OFF \
                       -DSPDLOG_FMT_EXTERNAL=ON"
```

**Review by CvH — CHANGES_REQUESTED** [review]

File: packages/network/libevent/package.mk [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+# Copyright (C) 2020-present Team LibreELEC (https://libreelec.tv)
+
+PKG_NAME="libevent"
+PKG_VERSION="2.1.12-stable"
+PKG_SHA256="92e6de1be9ec176428fd2367677e61ceffc2ee1cb119035037a27d346b0403bb"
+PKG_LICENSE="BSB-3c"
+PKG_SITE="https://github.com/libevent/libevent"
+PKG_URL="https://github.com/libevent/libevent/releases/download/release-$PKG_VERSION/libevent-$PKG_VERSION.tar.gz"
+PKG_LONGDESC="Event notification library"
+PKG_DEPENDS_TARGET="toolchain"
```
Comment (CvH) [inline packages/network/libevent/package.mk:11]:
```suggestion
PKG_DEPENDS_TARGET="toolchain openssl"
```

**Comment by CvH** [comment]
also pls change `libevent: new package 2.1.12-stable and used by avahi` to `libevent: initial commit` that it fits in house style

File: packages/network/libevent/package.mk (hunk 5 above)
Comment (CvH) [inline packages/network/libevent/package.mk:11]:
```suggestion
PKG_DEPENDS_TARGET="toolchain openssl"
```

**Review by CvH — CHANGES_REQUESTED** [review]

**Review by CvH — APPROVED** [review]

**Comment by lrusak** [comment]
What happened to using `--disable-libevent`??

File: packages/network/avahi/package.mk (hunk 1 above)
Comment (heitbaum) [reply packages/network/avahi/package.mk:11]: Hi @lrusak I have reviewed the avahi code and pull request https://github.com/lathiat/avahi/pull/186 which added libevent main loop adapter to avahi. I created a draft commit https://github.com/LibreELEC/LibreELEC.tv/commit/def3d0791fa05de4b578ad631e10fcd4c0d3fcfb and tested avahi build without libevent, but will complete a full build before the PR.

Comment (heitbaum) [reply packages/network/avahi/package.mk:11]: Completed code review, testing  and raised PR #4829

**Pull request merged**